x-pack/legacy/plugins/siem/public/pages/detection_engine/components/signals_info/index.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiLoadingSpinner } from '@elastic/eui';
+import { FormattedRelative } from '@kbn/i18n/react';
+import React, { useState, useEffect } from 'react';
+
+import { useQuerySignals } from '../../../../containers/detection_engine/signals/use_query';
+import { buildlastSignalsQuery } from './query.dsl';
+import { Aggs } from './types';
+
+interface SignalInfo {
+  ruleId?: string | null;
+}
+
+type Return = [React.ReactNode, React.ReactNode];
+
+export const useSignalInfo = ({ ruleId = null }: SignalInfo): Return => {
+  const [lastSignals, setLastSignals] = useState<React.ReactElement | null>(
+    <EuiLoadingSpinner size="m" />
+  );
+  const [totalSignals, setTotalSignals] = useState<React.ReactElement>(
+    <EuiLoadingSpinner size="m" />
+  );
```

Comment:
Talk about a compartmentalized and full featured little component here. Thanks for including the loader -- makes this easily used on the other pages. 👍